core/src/main/java/de/erethon/commons/misc/BlockUtil.java

```diff
@@ -166,4 +169,147 @@ public static Block getAttachedBlock(Block block) {
         return blocks;
     }
 
+    /**
+     * @param block the block to check
+     * @return true if the block is an Sign or WallSign, false otherwise
+     */
+    public static boolean isSignOrWallSign(Block block) {
+        return isSign(block) | isWallSign(block);
+    }
+
+    /**
+     * @param block the block to check
+     * @return true if the block is an Sign, false otherwise
+     */
+    public static boolean isSign(Block block) {
+        return block.getState() instanceof Sign;
+    }
+
+    /**
+     * @param block the block to check
+     * @return true if the block is an WallSign, false otherwise
+     */
+    public static boolean isWallSign(Block block) {
+        return block.getBlockData() instanceof WallSign;
+    }
+
+    /**
+     * This class stores an Location paired with an BlockFace.
+     * See usage down below.
+     *
+     * @see BlockUtil#getSignAttachedTo(Block)
+     * @see BlockUtil#getSignsAttachedTo(Block)
+     * @see BlockUtil#signAttachedTo(Block)
+     */
+    private static class LocationNode {
+        private final Location loc;
+        private final BlockFace face;
+
+        public LocationNode(Location loc, BlockFace face) {
+            this.loc = loc;
+            this.face = face;
+        }
+
+        public Location getLoc() {
+            return loc;
+        }
+
+        public BlockFace getFace() {
+            return face;
+        }
+    }
+
+    /**
+     * Returns the attached sign if found, null otherwise.
+     *
+     * @param block the block to check
+     * @return the attached sign if found, null otherwise
+     * @deprecated this method is not reliable if more than one sign is attached to the block.
+     * It will only return one of the attached signs, randomly picked and not specifically chosen.
+     */
+    @Nullable
+    @Deprecated
+    public static Sign getSignAttachedTo(Block block) {
```

Comment:
Ich würde es für sinnvoller halten, nur #getSignsAttachedTo zu haben. Das Set hat dann halt nur ein Schild in diesem Fall - aber zumindest ist es nicht zufällig, was eigentlich absolut nie hilfreich ist.